src/ng/plugins/visibility/visibility.js

```diff
@@ -14,15 +15,49 @@ TemplatePath
 class Visibility extends PluginComponent('qgrid.plugins.visibility.tpl.html') {
 	constructor() {
 		super(...arguments);
+		this.toggle = new Command({
+			execute: (key, parentKey = null) => {
+				if (parentKey) {
```

Comment:
it won't work with more complex objects
you need to close in Node setter/getter for visibility